### PR TITLE
debug: add thread id to log

### PR DESCRIPTION
**Pull request by halleyzhao**
- add tid in log
- add thread name for self created threads
- include common.mk for vpp

### CODE REVIEW

File: common/log.h [hunk 1]
```diff
@@ -38,6 +38,10 @@
 #include <stdint.h>
 #include <assert.h>
 #include <string.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#define GETTID()    syscall(__NR_gettid)
```
Comment (xuguangxin) [inline common/log.h]: will this cause problem for pure linux?

Comment (halleyzhao) [inline common/log.h]: it has been tested on ubuntu.

**Comment by xuguangxin** [comment]
ok, +1, thanks

**Comment by xuguangxin** [comment]
merged as 1ab59bcd12a3db458fe8b25852e9d2258bfbd4bf

**Pull request closed without merging**